src/main/java/io/vertx/core/http/impl/Http1xClientConnection.java

```diff
@@ -389,6 +395,14 @@ public void beginRequest(HttpClientRequestImpl req) {
           Object reqMetric = conn.metrics.requestBegin(conn.endpointMetric, conn.metric(), conn.localAddress(), conn.remoteAddress(), request);
           request.metric(reqMetric);
         }
+        VertxTracer tracer = context.tracer();
+        if (tracer != null) {
+          List<Map.Entry<String, String>> tags = new ArrayList<>();
+          tags.add(new AbstractMap.SimpleEntry<>("http.url", req.absoluteURI()));
+          tags.add(new AbstractMap.SimpleEntry<>("http.method", req.method().name()));
+          BiConsumer<String, String> headers = (key, val) -> req.headers().add(key, val);
+          trace = tracer.sendRequest(request.stream.getContext(), request, request.method.name(), headers, tags);
```

Comment:
this is presumptive about the data that will be parsed. Why not allow the impl to decide vs partially parsing here?

Reply:
you mean the `http.url`, etc... names ?

Reply:
not just the names.. that they are recorded at all. if you have access to req at the tracer impl, you can decide what to record.

Reply:
@adriancole so that was choice we made initially, however that forces the tracer to be cast the request object to `HttpClientRequest` which is fine.

However we can to have the `VertxTracer` to be used by other parts of the Vert.x stack and it means then that the tracer implementation would depend (at the artifact level) on too many artifacts and have specific code for each of them. This choice allows to decouple the tracing implementation from the activity being traced.